### PR TITLE
Issue230 a setup

### PR DESCRIPTION
**Pull request by ekr**
@juberti I don't think this is quite ready, but PTAL

### CODE REVIEW

File: draft-ietf-rtcweb-jsep.xml [hunk 1]
```diff
@@ -2636,12 +2639,17 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
               algorithm used for the fingerprint MUST match that used
               in the certificate signature.</t>
 
+              <t>An "a=dtls-id" line, as specified in
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: Suggest matching the offer text and putting this after a=setup

File: draft-ietf-rtcweb-jsep.xml [hunk 2]
```diff
@@ -2996,6 +3004,11 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
             <xref target="RFC4145" />, Section 4, and the setup value
             is stored.</t>
 
+            <t>If present, a single "a=dtls-id" line is parsed as
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: Is this thing valid at session level?

File: draft-ietf-rtcweb-jsep.xml [hunk 3]
```diff
@@ -3199,12 +3212,20 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
               the size limits specified in
               <xref target="RFC5245" />, Section 15.4.</t>
 
+              <t>dtls-id value, which MUST be set according to
+              <xref target="I-D.ietf-mmusic-dtls-sdp"/> Section 5.
+              If this is a re-offer and the dtls-id value is different
+              from that presently in use, then a new DTLS connection
+              MUST be established, and the ufrag and password values
+              MUST also be different. If processing an answer, the
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: I would spell this out, i.e., if we're doing a new DTLS connection here, it MUST be as part of a proposed ICE restart, with new ICE ufrag and password values.

File: draft-ietf-rtcweb-jsep.xml [hunk 4]
```diff
@@ -3199,12 +3212,20 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
               <t>DTLS setup value, which MUST be set according to the
-              rules specified in
-              <xref target="RFC5763" />, Section 5, and MUST be
-              consistent with the selected role of the current DTLS
-              connection, if one exists.[TODO: may need revision, i.e.,
-              use of actpass</t>
+              rules specified in <xref target="RFC5763"/>,
+              Section 5 and MUST
+              be consistent with the selected role of the current DTLS
+              connection, if one exists and is being continued.</t>
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: I also think you need some text in the "applying a remote description" section below that talks about creating a new DTLS connection.

File: draft-ietf-rtcweb-jsep.xml [hunk 5]
```diff
@@ -3199,12 +3212,20 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
               the size limits specified in
               <xref target="RFC5245" />, Section 15.4.</t>
 
+              <t>dtls-id value, which MUST be set according to
+              <xref target="I-D.ietf-mmusic-dtls-sdp"/> Section 5.
+              If this is a re-offer and the dtls-id value is different
+              from that presently in use, then a new DTLS connection
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: Not sure if this is where you need to point out that a new DTLS connection MUST be established - that seems like something for the  "applying a remote desc" section. I think you just need to note that the previous connection is not being continued, relevant to the para below.

File: draft-ietf-rtcweb-jsep.xml (hunk 2 above)
Comment (ekr) [inline draft-ietf-rtcweb-jsep.xml]: Thankfully, no
https://tools.ietf.org/html/draft-ietf-mmusic-dtls-sdp-14#section-4

File: draft-ietf-rtcweb-jsep.xml (hunk 3 above)
Comment (ekr) [inline draft-ietf-rtcweb-jsep.xml]: done

**Comment by ekr** [comment]
@juberti PTAL

File: draft-ietf-rtcweb-jsep.xml (hunk 2 above)
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: Did you mean to remove this then?

File: draft-ietf-rtcweb-jsep.xml [hunk 6]
```diff
@@ -3199,12 +3212,21 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
               the size limits specified in
               <xref target="RFC5245" />, Section 15.4.</t>
 
+              <t>dtls-id value, which MUST be set according to
+              <xref target="I-D.ietf-mmusic-dtls-sdp"/> Section 5.
+              If this is a re-offer and the dtls-id value is different
+              from that presently in use, the DTLS connection is not being
+              continued and the remote description MUST be part of an
+              ICE restart, together with new ufrag and password
+              values. If processing an answer, the
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: suggest changing 'processing' to 'this is' to match text above

File: draft-ietf-rtcweb-jsep.xml [hunk 7]
```diff
@@ -3389,6 +3411,17 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
           the end-of-candidates indication as described in
           <xref target="I-D.ietf-ice-trickle" /> Section 11.</t>
 
+          <t>If the dtls-id has changed from the previous remote description,
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: Looking at this more closely, I think you probably want to move this text to the 'applying an answer' section, which already has a para on when to create a DTLS connection.

**Pull request merged**